js/interactive-guides/step-content.js

```diff
@@ -153,20 +153,37 @@ var stepContent = (function() {
       return;
     }
     var description = step.description;    
-    var jointDescription = description;
     if ($.isArray(description)) {
-      jointDescription = description.join("<br/>");
+        $.each(description, function(i, desc) {
+            if (desc) {
+                //TODO: check for header or <li> tag and ignore these
+                if (__containsHTMLTag(desc)) {
+                } else {
+                    description[i] = $('<p>').html(desc).prop('outerHTML'); //Use .text instead of .html for debugging
+                }
+            }
+        });
+        description = description.join("");
     }
     var newDescription = $("<div class='description' tabindex='0'></div>");
     newDescription.attr('data-step', step.name);
-    newDescription.html(jointDescription);
+    newDescription.html(description);
     $("#contentContainer").append(newDescription);
 
     if (step.name === "RelateGuides") {
       insertRelateGuidesContent();  
     }
   };
 
+  var __containsHTMLTag = function(content) {
```

Comment:
It is confusing why we only checked for certain tags if the method is called containsHTMLTag(). We should at least have a comment indicating why we are checking particular tags instead of checking if anything in it is html.